sycl/unittests/Extensions/USMMemcpy2D.cpp

```diff
@@ -0,0 +1,651 @@
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+    };
+    return DummySignature[Idx];
+  }
+  static constexpr bool isESIMD() { return false; }
+  static constexpr bool callsThisItem() { return false; }
+  static constexpr bool callsAnyThisFreeFunction() { return false; }
+  static constexpr int64_t getKernelSize() {
+    return 2 * sizeof(void *) + 2 * sizeof(sycl::id<2>) + 4 * sizeof(size_t);
+  }
+};
+
+template <> struct KernelInfo<class __usmmemcpy2d<unsigned char>> {
+  static constexpr const char *getName() {
+    return USMMemcpyHelperKernelNameChar;
+  }
+  static constexpr unsigned getNumParams() { return 8; }
+  static const kernel_param_desc_t &getParamDesc(int Idx) {
+    // Actual signature does not matter, but we need entries for each param.
+    static constexpr const kernel_param_desc_t DummySignature[] = {
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+        {kernel_param_kind_t::kind_std_layout, 4, 0},
+    };
+    return DummySignature[Idx];
+  }
+  static constexpr bool isESIMD() { return false; }
+  static constexpr bool callsThisItem() { return false; }
+  static constexpr bool callsAnyThisFreeFunction() { return false; }
+  static constexpr int64_t getKernelSize() {
+    return 2 * sizeof(void *) + 2 * sizeof(sycl::id<2>) + 4 * sizeof(size_t);
+  }
+};
+} // namespace detail
+} // __SYCL_INLINE_VER_NAMESPACE(_V1)
+} // namespace sycl
+
+static sycl::unittest::PiImage generateMemopsImage() {
+  using namespace sycl::unittest;
+
+  PiPropertySet PropSet;
+
+  std::vector<unsigned char> Bin{10, 11, 12, 13, 14, 15}; // Random data
+
+  PiArray<PiOffloadEntry> Entries = makeEmptyKernels(
+      {USMFillHelperKernelNameLong, USMFillHelperKernelNameChar,
+       USMMemcpyHelperKernelNameLong, USMMemcpyHelperKernelNameChar});
+
+  PiImage Img{PI_DEVICE_BINARY_TYPE_SPIRV,            // Format
+              __SYCL_PI_DEVICE_BINARY_TARGET_SPIRV64, // DeviceTargetSpec
+              "",                                     // Compile options
+              "",                                     // Link options
+              std::move(Bin),
+              std::move(Entries),
+              std::move(PropSet)};
+
+  return Img;
+}
+
+sycl::unittest::PiImage Imgs[] = {generateMemopsImage()};
```

Comment:
I suggest that we wrap this into an anonymous namespace to avoid potential clashes with other tests

Reply:
I've put them in an anonymous namespace. Was this what you were thinking?

Reply:
Exactly, thanks!